### PR TITLE
Replace Draftail with EditorJS

### PR DESCRIPTION
**Pull request by dominik-zeglen**
I want to merge this change because it improves rich text editor 🎉 

**PR intended to be tested with API branch:** master

### Pull Request Checklist

1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Translated strings are extracted.
1. [ ] Number of API calls is optimized.
1. [ ] The changes are tested.
1. [ ] Data-test are added for new elements.
1. [ ] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-916 Switch CMS to Editor.js](https://app.clickup.com/t/2549495/SALEOR-916)

File: src/collections/components/CollectionCreatePage/CollectionCreatePage.tsx [hunk 1]
```diff
@@ -2,64 +2,30 @@ import AppHeader from "@saleor/components/AppHeader";
...
-}
+import CollectionCreateForm, { CollectionCreateData } from "./form";
 
 export interface CollectionCreatePageProps {
   disabled: boolean;
   errors: ProductErrorFragment[];
   saveButtonBarState: ConfirmButtonTransitionState;
   onBack: () => void;
-  onSubmit: (data: CollectionCreatePageFormData) => void;
+  onSubmit: (data: CollectionCreateData) => Promise<any[]>;
```
Comment (orzechdev) [inline src/collections/components/CollectionCreatePage/CollectionCreatePage.tsx:26]:
```suggestion
  onSubmit: (data: CollectionCreateData) => SubmitPromise;
```

File: src/collections/components/CollectionCreatePage/form.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  change: FormChange;
+  data: CollectionCreateData;
+  handlers: CollectionCreateHandlers;
+  hasChanged: boolean;
+  submit: () => Promise<boolean>;
+}
+
+export interface CollectionCreateFormProps {
+  children: (props: UseCollectionCreateFormResult) => React.ReactNode;
+  onSubmit: (data: CollectionCreateData) => Promise<any[]>;
```
Comment (orzechdev) [inline src/collections/components/CollectionCreatePage/form.tsx:41]:
```suggestion
  onSubmit: (data: CollectionCreateData) => SubmitPromise;
```

File: src/collections/components/CollectionCreatePage/form.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+function useCollectionCreateForm(
+  onSubmit: (data: CollectionCreateData) => Promise<any[]>
```
Comment (orzechdev) [inline src/collections/components/CollectionCreatePage/form.tsx:45]:
```suggestion
  onSubmit: (data: CollectionCreateData) => SubmitPromise
```

File: src/collections/components/CollectionDetailsPage/CollectionDetailsPage.tsx [hunk 4]
```diff
@@ -51,7 +35,7 @@ export interface CollectionDetailsPageProps extends PageListProps, ListActions {
   onImageDelete: () => void;
   onImageUpload: (file: File) => void;
   onProductUnassign: (id: string, event: React.MouseEvent<any>) => void;
-  onSubmit: (data: CollectionDetailsPageFormData) => void;
+  onSubmit: (data: CollectionUpdateData) => Promise<any>;
```
Comment (orzechdev) [inline src/collections/components/CollectionDetailsPage/CollectionDetailsPage.tsx:38]:
```suggestion
  onSubmit: (data: CollectionUpdateData) => SubmitPromise;
```

File: src/pages/components/PageDetailsPage/form.tsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  data: PageData;
+  handlers: PageUpdateHandlers;
+  hasChanged: boolean;
+  submit: () => void;
+}
+
+export interface PageFormProps {
+  children: (props: UsePageUpdateFormResult) => React.ReactNode;
+  page: PageDetails_page;
+  onSubmit: (data: PageData) => Promise<any[]>;
```
Comment (orzechdev) [inline src/pages/components/PageDetailsPage/form.tsx:41]:
```suggestion
  onSubmit: (data: PageData) => SubmitPromise;
```

File: src/pages/components/PageDetailsPage/form.tsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+function usePageForm(
+  page: PageDetails_page,
+  onSubmit: (data: PageData) => Promise<any[]>
```
Comment (orzechdev) [inline src/pages/components/PageDetailsPage/form.tsx:46]:
```suggestion
  onSubmit: (data: PageData) => SubmitPromise
```

**Review by orzechdev — CHANGES_REQUESTED** [review]

**Review by orzechdev — APPROVED** [review]

**Pull request merged**